src/RingGrids/interpolation.jl

```diff
@@ -119,14 +135,18 @@ expected to have two fields,
     
 NF is the number format used to calculate the interpolation, which can be
 different from the input data and/or the interpolated data on the new grid."""
-abstract type AbstractInterpolator{NF, G} end
+abstract type AbstractInterpolator end
 
-struct AnvilInterpolator{NF<:AbstractFloat, G<:AbstractGrid} <: AbstractInterpolator{NF, G}
-    geometry::GridGeometry{G}
-    locator::AnvilLocator{NF}
+struct AnvilInterpolator{NF, Grid, Geometry, Locator} <: AbstractInterpolator
+    geometry::Geometry
+    locator::Locator
 end
 
-function Base.show(io::IO,L::AnvilInterpolator{NF, Grid}) where {NF, Grid}
+eltype(::AnvilInterpolator{NF}) where NF = NF
```

Comment:
Did you mean to extend `Base.eltype`?
```suggestion
Base.eltype(::AnvilInterpolator{NF}) where NF = NF
```

Reply:
touché!